src/users/users.service.ts

```diff
@@ -15,7 +15,10 @@ export class UsersService {
   ) {}
 
   async getLoginHistories(id: string) {
-    const user = await this.usersRepo.findOne({ where: { id: id } });
+    const user = await this.usersRepo.findOne({
+      where: { id: id },
```

Comment:
Why don't you just query the history repo?
Also wrap your async code in try catch